app/src/org/commcare/dalvik/activities/CommCareHomeActivity.java

```diff
@@ -1543,4 +1381,25 @@ public boolean isBackEnabled() {
         return false;
     }
 
+    /**
+     * For Testing purposes only
+     */
+    public SessionNavigator getSessionNavigator() {
+        if (BuildConfig.DEBUG) {
+            return sessionNavigator;
+        } else {
+            throw new RuntimeException("On principal of design, only meant for testing purposes");
+        }
+    }
+
+    /**
+     * For Testing purposes only
+     */
+    public void setFormAndDataSyncer(FormAndDataSyncer formAndDataSyncer) {
+        if (BuildConfig.DEBUG) {
+            this.formAndDataSyncer = formAndDataSyncer;
+        } else {
+            throw new RuntimeException("On principal of design, only meant for testing purposes");
+        }
```

Comment:
Don't have any great ideas for an alternative. The way I've made this sort of situation a little bit cleaner in the past is to make the object that has a test version be an argument to the constructor, and pass in the appropriate version for each ase. But since this is an activity that's not really an option. Seems ok to leave like this for now.